plugins/uefi-dbx/fu-plugin-uefi-dbx.c

```diff
@@ -0,0 +1,94 @@
+/*
+ * Copyright (C) 2020 Richard Hughes <richard@hughsie.com>
+ *
+ * SPDX-License-Identifier: LGPL-2.1+
+ */
+
+#include "config.h"
+
+#include "fu-plugin-vfuncs.h"
+#include "fu-efivar.h"
+#include "fu-hash.h"
+#include "fu-uefi-dbx-common.h"
+#include "fu-uefi-dbx-file.h"
+
+struct FuPluginData {
+	gchar			*fn;
+};
+
+void
+fu_plugin_init (FuPlugin *plugin)
+{
+	fu_plugin_alloc_data (plugin, sizeof (FuPluginData));
+	fu_plugin_set_build_hash (plugin, FU_BUILD_HASH);
+}
+
+void
+fu_plugin_destroy (FuPlugin *plugin)
+{
+	FuPluginData *data = fu_plugin_get_data (plugin);
+	g_free (data->fn);
+}
+
+gboolean
+fu_plugin_startup (FuPlugin *plugin, GError **error)
+{
+	FuPluginData *data = fu_plugin_get_data (plugin);
+	GPtrArray *checksums;
+	gsize bufsz = 0;
+	guint missing_cnt = 0;
+	g_autofree guint8 *buf_system = NULL;
+	g_autofree guint8 *buf_update = NULL;
+	g_autoptr(FuUefiDbxFile) dbx_system = NULL;
+	g_autoptr(FuUefiDbxFile) dbx_update = NULL;
+	g_autoptr(GError) error_local = NULL;
+
+	/* get binary blob */
+	data->fn = fu_uefi_dbx_get_dbxupdate (error);
+	if (data->fn == NULL) {
+		g_prefix_error (error,
+				"file can be downloaded from %s: ",
+				FU_UEFI_DBX_DATA_URL);
```

Comment:
If we aren't offering a fwupdtool/fwupdmgr command that downloads it, we probably should put in the message where the user needs to store it.  Or distribute a helper script that does it with fwupd.

Reply:
> where the user needs to store it

Reply:
Yeah I think that makes sense.

Reply:
I did the /var/lib/fwupd/dbx thing. At the moment the error is just a console warning, but with the hsi branch it's visible in `fwupdmgr security`